ingestion/src/metadata/interfaces/sqa_interface.py

```diff
@@ -127,7 +131,27 @@ def _session_factory(service_connection_config):
         garbage collected once the application thread ends
         """
         engine = get_connection(service_connection_config)
-        return create_and_bind_thread_safe_session(engine)
+        session = create_and_bind_thread_safe_session(engine)
+        SQAInterface.set_session_tag(service_connection_config, engine)
+        return session
+
+    @staticmethod
+    def set_session_tag(service_connection_config, engine):
```

Comment:
We'll need to pass the `session` instance here (vs `engine`) and call that function l. 95 and l. 255

Reply:
@TeddyCr, we are calling this function inside session factory, do we still need to do that then 🤔 ?